### PR TITLE
Logging infrastructure

### PR DESCRIPTION
**Pull request by mts299**
This PR is the initial push to develop. It includes the initial setup of pydarn including logging. 

@kkotyk Can you just review the logging_config.yaml, setup.py, pydarn/__initi__.py and the small changes I made to dmap.py for logging. 

I will be looking at dmap.py and the README in other PR's.

### CODE REVIEW

**Comment by mts299** [comment]
Yes, I forgot I added the fitacf and map file into the test to try and get your tests to work. I will look into trimming them on the next commit. Definitely, something I will look at for the PR on dmap.py

File: pydarn/exceptions/pydmap_exceptions.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                " is currupt because: {msg}".format(filename=filename,
+                                                    msg=message)
+        super().__init__(self.message)
+        pydarn_logger.error("DmapDataError: {} is corrupt {}".format(filename,
+                                                                     message))
+
+
+class CorruptDataError(Exception):
+    """
+    Raise when data is curropt in the file
```
[the remark this answers is not in the document]
Comment (mts299) [reply pydarn/exceptions/pydmap_exceptions.py]: I should really look into a spellchecker for vim. Done

File: pydarn/logging_config.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#
+# logging_config.yaml 
+# 2018-11-05
+# Configuration for pydarn's logging.
+
+# author: 
+# Marina Schmidt 
+
+# version number needs to be 1, cannot start at 0 
+version: 1
```
[the remark this answers is not in the document]
Comment (mts299) [reply pydarn/logging_config.yaml]: Done, added more details and included two links. One explaining Yaml and the other is a logging tutorial for python to explain the basics. This information will also be further talked about in the README for developers.

**Comment by ksterne** [comment]
Or is it possible to have some kind of public web access for these files?  I know that gets tricky, but there's got to be some kind of google drive or other file share hosting that we could put small subsets of data onto that could be accessed.  Or maybe that's not possible as it needs to have the files locally.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+running the files.
+
+author:
+Marina Schmidt
+"""
+
+from distutils.core import setup
+from setuptools import setup, find_packages
+
+# currently not implemented due to some challenges
```
[the remark this answers is not in the document]
Comment (mts299) [reply setup.py]: Done

File: test/dmap_tests/pydarn.log [hunk 4]
```diff
@@ -0,0 +1,276 @@
+2018-11-05 14:03:15,528 pydarn: READ DATA: cursor 0 bytelen 12190580
```
[the remark this answers is not in the document]
Comment (mts299) [reply test/dmap_tests/pydarn.log]: I added *.log in the git ignore so this does not happen again.

File: test/dmap_tests/pydarn_errors.log [hunk 5]
```diff
@@ -0,0 +1,20 @@
+2018-11-05 15:49:33,395 pydarn: ERROR DmapDataError: testfiles/20160301.C0.sas.fitacf is corrupt INITIAL INTEGRITY: Initial integrity check shows
```
[the remark this answers is not in the document]
Comment (mts299) [reply test/dmap_tests/pydarn_errors.log]: Removed.

**Comment by mts299** [comment]
There is a git tool that allows you to store larger files... that we could use. I had to install it one time when I committed large files:
https://git-lfs.github.com/

**Comment by egthomas** [comment]
Sorry, I haven't been following the activity on this repository.  Didn't we create the [rst-test](https://github.com/SuperDARN/rst-test) repository sometime in the summer for the purpose of storing test data?  Should it be named something more generic instead?

**Comment by kkotyk** [comment]
@egthomas Yes we did. And we should be using that more. In this case when I created this code I made unit tests for testing dmap and used these files for testing. In reality, all thats required here are a few trimmed records. These tests aren't for scientific results, but rather just to see if pydmap works correctly after modifications. For true correctness, this code should be tested against the test data set as a final check.

**Comment by mts299** [comment]
@egthomas No worries this is the first activity pydarn has had so you haven't missed much. 

That is what I was thinking when @ksterne brought this up. Maybe rename it to something like "data-test-set"?

The test files, in this case, happened due to some copying and git adding with me forgetting about the big files.

**Pull request merged**